File: fizzBuzz.js
// function fizzBuzz(n) {
//   for (let i = 1; i <= n; i++) {
//     if (i % 15 === 0) {
//       console.log("fizzbuzz");
//     } else if (i % 3 === 0) {
//       console.log("fizz");
//     } else if (i % 5 === 0) {
//       console.log("buzz");
//     } else {
//       console.log(i);
//     }
//   }
// }

function fizzBuzz(n) {
  let msg;
  for (let i = 1; i <= n; i++, msg = "") {
    if (!(i % 3)) msg += "fizz";
    if (!(i % 5)) msg += "buzz";
    console.log(msg || i);
  }
}

fizzBuzz(22);

File: fizz_buzz.rb
def fizz_buzz(n)
  1.upto(n).each do |num|
    if num % 3 == 0
      if num % 5 == 0
        p "fizzbuzz"
      else
        p "fizz"
      end
    elsif num % 5 == 0
      p "buzz"
    else
      p num
    end
  end
end

fizz_buzz(22)
